### PR TITLE
feat(e2e): add e2e tests for the evidence module 

### LINKED ISSUE #1676: Add Evidence E2E tests
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by Pantani**
close #1676

## Description

Since the evidence module doesn't have gaiad tx, this PR only adds evidence into the genesis JSON and queries all evidence.

### CODE REVIEW

File: tests/e2e/e2e_evidence_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/x/evidence/exported"
+	evidencetypes "github.com/cosmos/cosmos-sdk/x/evidence/types"
+)
+
+func (s *IntegrationTestSuite) TestEvidence() {
+	s.Run("teste query evidences", func() {
```
Comment (mmulji-ic) [inline tests/e2e/e2e_evidence_test.go]: teste query evidences -> test evidence queries

File: tests/e2e/e2e_exec_test.go [hunk 2]
```diff
@@ -602,6 +603,28 @@ func (s *IntegrationTestSuite) registerICA(owner, connectionID string) {
 	s.T().Logf("%s reigstered an interchain account on chain %s from chain %s", owner, s.chainB.id, s.chainA.id)
```
Comment (mmulji-ic) [inline tests/e2e/e2e_exec_test.go]: spelling - registered

File: tests/e2e/e2e_exec_test.go [hunk 3]
```diff
@@ -602,6 +603,28 @@ func (s *IntegrationTestSuite) registerICA(owner, connectionID string) {
...
+	s.T().Logf("querying evidence %s on chain %s", hash, c.id)
+
+	gaiaCommand := []string{
+		gaiadBinary,
+		"q",
+		"evidence",
+		hash,
+	}
+
+	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, func(stdOut []byte, stdErr []byte) bool {
```
Comment (mmulji-ic) [inline tests/e2e/e2e_exec_test.go]: Thxs for filing with sdk folks!

Comment (glnro) [inline tests/e2e/e2e_exec_test.go]: Nit: we might want to rename or rework `executeGaiaTxCommand` for cli commands that don't require validation of the tx hash...`keys` and `query` subcommands would fall under this category.

File: tests/e2e/e2e_evidence_test.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package e2e
```
Comment (mmulji-ic) [inline tests/e2e/e2e_evidence_test.go:1]: A more complicated scenario would be to have blocks with the following:
- DuplicateVoteEvidence
- LightClientAttackEvidence
... and check for those as well. Future todo?

File: tests/e2e/e2e_exec_test.go (hunk 3 above)
Comment (Pantani) [reply tests/e2e/e2e_exec_test.go]: Yes, I agree! I also moved the other API queries to the `querie.go` file in this PR #1780. The evidence API seems not to be working well. I opened an issue in the cosmos-sdk repo (https://github.com/cosmos/cosmos-sdk/issues/13444). We can only query by CLI now, my idea was to try to put all CLI exec commands in the same file, but we can move to `e2e_query_test.go`. WDYT?

Comment (glnro) [reply tests/e2e/e2e_exec_test.go]: yeah sounds great 👍

**Review by okwme — APPROVED** [review]
LGTM! Feel free to merge @Pantani : )

**Pull request merged**